cyclops-ui/src/components/pages/Modules/Modules.tsx

```diff
@@ -45,18 +54,50 @@ const Modules = () => {
         setLoadingModules(false);
       });
   }, []);
+  useEffect(() => {
+    var updatedList = [...allData];
+    updatedList = updatedList.filter((module: any) => {
+      return (
+        module.name.toLowerCase().indexOf(searchInputFilter.toLowerCase()) !==
+        -1
+      );
+    });
+    const newfilteredData = updatedList.filter((module: any) =>
+      moduleHealthFilter
+        .map((status) => status.toLowerCase())
+        .includes(module.status.toLowerCase()),
+    );
+
+    if (JSON.stringify(newfilteredData) !== JSON.stringify(filteredData)) {
+      setFilteredData(newfilteredData);
+    }
+  }, [moduleHealthFilter, filteredData, allData, searchInputFilter]);
```

Comment:
@nikhiltanna, can you remove `filteredData` from the dependencies?

Reply:
Done please check if it is working fine now

Reply:
Should I add the dependency or silent it??

Reply:
Even if we keep filerdata as dependency it's will not  affect as I have kept if conditions to update the filterdata  both ways  it would work actually

Reply:
@nikhiltanna Actually, if you remove the condition (the one with JSON.stringify) and just set filtered data you should be able to remove filteredData from dependencies. I prefer we don't have filteredData there because each time you change it, it will trigger the useEffect for nothing

Reply:
Done